.github/workflows/build-docker.yml

```
@@ -0,0 +1,115 @@
+name: Build and Package
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main", "dev", "github-action-docker-image" ]
+
+jobs:
+  build_docker_image:
+    name: Build Docker Image for CI
+    if: github.repository_owner == 'CTSRD-CHERI'
+    runs-on: ubuntu-24.04
+    permissions:
+      contents: read
+      packages: write
+      checks: write
+      pull-requests: write
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Free up disk space
+      uses: jlumbroso/free-disk-space@main
+      with:
+        # We run out of disk space when building all targets and docker images
+        # Use this action to delete some SDKs that we do not require.
+        android: true
+        dotnet: true
+        haskell: true
+        large-packages: false
+        swap-storage: false
+        docker-images: false
+
+    - name: Install Dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y ninja-build cmake build-essential lld
+
+    - name: ccache
+      uses: hendrikmuhs/ccache-action@v1.2
+
+    - name: Configure CMake
+      run: |
+        # Build a minimal version of LLVM with features we do not need disabled
+        cmake -G Ninja -S llvm -B build \
+          -DLLVM_ENABLE_ASSERTIONS=TRUE \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_INSTALL_PREFIX=/usr/local \
+          -DLLVM_ENABLE_PROJECTS="clang;lld" \
+          -DLLVM_TARGETS_TO_BUILD="all" \
+          -DLLVM_CCACHE_BUILD=TRUE \
+          -DLLVM_ENABLE_LLD=TRUE \
+          -DCLANG_ENABLE_STATIC_ANALYZER=FALSE -DCLANG_ENABLE_ARCMT=FALSE -DLLVM_ENABLE_Z3_SOLVER=FALSE \
+          -DLLVM_TOOL_LLVM_MCA_BUILD=FALSE -DLLVM_TOOL_LLVM_EXEGESIS_BUILD=FALSE -DLLVM_TOOL_LLVM_RC_BUILD=FALSE \
+          -DLLVM_LIT_ARGS="-v --xunit-xml-output=test-results.xml"
+
+    - name: Build
+      run: |
+        df -h
+        cmake --build build
+        df -h
+
+    - name: Run Tests
+      run: |
+        cmake --build build --target check-all
+        df -h
+
+    - name: Publish Test Results
+      uses: EnricoMi/publish-unit-test-result-action@v2
+      if: always()
```

Does this work ok if the build itself fails?

---

```suggestion
      if: (!cancelled())
```

Apparently this is the recommended approach on the actions readme. It is needed so that the test result get published even if the test step returns non-zero.